src/models/clap.py

```diff
@@ -0,0 +1,38 @@
+import torch
+from .audio import BaseAudioEncoder
+from .text import BaseTextEncoder
+from typing import Union, List, Dict
+import numpy as np
+
+
+class BaseClap(torch.nn.Module):
+    def __init__(
+            self,
+            audio_encoder: BaseAudioEncoder,
+            text_encoder: BaseTextEncoder
+    ):
+        super().__init__()
+        self.audio_encoder = audio_encoder
+        self.text_encoder = text_encoder
+        self.loss_fn = torch.nn.CrossEntropyLoss()
+
+    def forward(self, texts: Union[List[str], str], audios: Union[List[np.array], np.array]) -> Dict[str, torch.Tensor]:
+        """
+        :param texts: texts for the input
+        :param audios: audios for the
+        :return: dictionary with cosine distance between embeddings
+            if len(texts) == len(audios) also returns the loss
+        """
+        num_texts = 1 if isinstance(texts, str) else len(texts)
+        num_audios = 1 if isinstance(audios, str) else len(audios)
```

Comment:
better do something like it in other method. Like we can have predict() that wraps object in the required shapes but forward should have normal sizes at start

Reply:
I thought about it, I can do it if you want, but I am not sure whether this will be usefull. HF processor for example can handle multiple inputs and it is convinient, that is why I did, what I have done

Reply:
yes, it can. But that's processor and not model. I just suspect it could be slow in training because we waste time. AFAIK HuggingFace models expect correct shapes always